cartridges/int_algolia/cartridge/scripts/algolia/lib/algoliaData.js

```diff
@@ -145,12 +146,30 @@ function getInstanceHostName() {
 }
 
 /**
- * Create index id for search results request
+ * Create indexID prefix for search results request
+ * If custom site preference Algolia_IndexIDPrefix is set in BM,
+ * its value will be used as a prefix instead of the first part of the hostname and the siteID
+ * @returns {string} indexID prefix
+ */
+function getIndexIDPrefix() {
+    var indexIDPrefix = getPreference('IndexIDPrefix');
+
+    if (!empty(indexIDPrefix)) {
+        return indexIDPrefix.trim();
+    } else {
+        return getInstanceHostName() + '__' + currentSite.getID();
+    }
+}
+
+/**
+ * Create indexID for search results request
+ * If custom site preference Algolia_IndexIDPrefix is set in BM,
+ * its value will be used as a prefix instead of the first part of the hostname and the siteID
  * @param {string} type - type of indices: products | categories
  * @returns {string} indexId
  */
 function calculateIndexId(type) {
```

Comment:
I guess this is because of this method name that you kept "ID" in the name? Let's forget about that name and go with "index prefix" everywhere?

Reply:
Done.